File: js/Strings/isPalindrome.js
//input: string, name
//output: boolean, true is input is palindrome, otherwise false

const isPalindrome = (word) => {
  if (typeof word !== "string") {
    return false;
  }
  const lowerCasedInput = word.toLowerCase();

  let start = 0;
  let end = lowerCasedInput.length - 1;

  while (start < end) {
    console.log("lowerCasedInput[start]: ", lowerCasedInput[start]);
    console.log("lowerCasedInput[end]: ", lowerCasedInput[end]);
    if (lowerCasedInput[start] !== lowerCasedInput[end]) {
      return false;
    }
    start++;
    end--;
  }
  return true;
};

console.log(isPalindrome("Hannah"));
console.log(isPalindrome("hannah"));
console.log(isPalindrome("foot"));
console.log(isPalindrome("f"));
console.log(isPalindrome("ft"));
console.log(isPalindrome(33));

File: python/Strings/isPalindrome.py
msg = "hello world"
print(msg)

#GOAL: determine whether input str is a palindrome
    #palindrome = same character at every opposite index so str[0] must === str[str.length - 1] all the way to the center of the str
# input: string
# output: boolean, true if string is a palindrome

def isPalindrom(str):
    start = 0
    end = len(str) - 1

    while start < end:
        if str[start] != str[end]:
            return False
        start += 1
        end -= 1

    return True

print(isPalindrom('hannah'))
print(isPalindrom('phillip'))